edx_exams/apps/api/serializers.py

```diff
@@ -40,6 +47,53 @@ class Meta:
         fields = ['name', 'verbose_name', 'lti_configuration_id']
 
 
+class CourseExamConfigurationSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the CourseExamConfiguration model
+    """
+    provider = serializers.CharField(source='provider.name', allow_null=True)
+
+    # The escalation_email is only required when the provider is not None.
+    # We enforce this constraint in the validate function.
+    escalation_email = serializers.EmailField(required=False, default='')
+
+    class Meta:
+        model = CourseExamConfiguration
+        fields = ['provider', 'escalation_email']
+
+    def validate_provider(self, value):
+        """
+        Validate that the provider value corresponds to an existing ProctoringProvider. If the value is None,
+        the value is valid.
+        """
+        if value is not None:
+            try:
+                ProctoringProvider.objects.get(name=value)
+            # This exception is handled by the Django Rest Framework, so we don't want to use raise from and risk
+            # getting the stack trace included in client facing errors.
+            except ProctoringProvider.DoesNotExist:
+                raise serializers.ValidationError(  # pylint: disable=raise-missing-from
+                    'Proctoring provider does not exist.'
+                )
+            return value
+
+        return value
+
+    def validate(self, attrs):
+        """
+        Validate that escalalation_email is provided when the provider is provided.
+
+        NOTE: Currently, the LTI-based proctoring providers will require an escalation_email. Because the
```

Comment:
Just to be sure we're on the same page, the empty string default is on the serializer. There is no default on the model, but blanks are allowed. Do you mean the default on the serializer? Or the blank on the model?

I changed the model field and forgot to update this comment, so that might be why I was talking about the model here.

If I don't define the `escalation_email` serializer field explicitly, the DRF-generated one will look like this.

```
CourseExamConfigurationSerializer():
    provider = CharField(allow_null=True, source='provider.name')
    escalation_email = EmailField(allow_blank=True, max_length=254, required=False)
```

If I remove the explicit `escalation_email` serializer field and remove this comment, does that work? I think a more accurate comment is, "Because an escalation email is only required when a proctoring provider is set, we have to enforce this requirement in the validate function."

Reply:
ah yeah I just got confused by the comment here and thought you were referring to the model itself. I'm still fuzzy on what the explicit `escalation_email` serializer field is doing for us. The default _looks_ fine to me but I might be missing some nuance here.